lib/google_contacts_api/group.rb

```diff
@@ -40,5 +40,18 @@ def edit_link
       _link = self["link"].find { |l| l.rel == "edit" }
       _link ? _link.href : nil
     end
+
+    def self.create(attrs, api)
+      response = api.post('groups/default/full', xml_for_create(attrs), {}, 'Content-Type' => 'application/atom+xml')
+      Group.new(Hashie::Mash.new(JSON.parse(response.body)).entry)
+    end
+
+    def self.xml_for_create(group_attrs)
+      @@new_group_template ||=
+        File.new(File.dirname(__FILE__) + '/templates/group.xml.erb').read
+
+      # Passes in group_attrs to the template via binding
+      ERB.new(@@new_group_template).result(binding)
```

Comment:
That having been said, it might be better to use a heredoc to load the template instead of doing a file read.